Framework/Avoidance/Handlers/AnimationCircularAvoidanceHandler.cs

```diff
@@ -1,39 +1,42 @@
 ﻿using System;
+using Trinity.Framework.Actors.ActorTypes;
 using Trinity.Framework.Avoidance.Structures;
 using Trinity.Framework.Grid;
+using Zeta.Bot.Navigation;
+using Zeta.Game.Internals.Actors.Gizmos;
 
 namespace Trinity.Framework.Avoidance.Handlers
 {
     public class AnimationCircularAvoidanceHandler : IAvoidanceHandler
     {
-        public void UpdateNodes(TrinityGrid grid, Structures.Avoidance avoidance)
+        public bool UpdateNodes(TrinityGrid grid, Structures.Avoidance avoidance)
         {
-            foreach (var actor in avoidance.Actors)
-            {
-                if (actor == null || !actor.IsValid)
-                    continue;
+            var actor = Core.Actors.RactorByRactorId<TrinityActor>(avoidance.RActorId);
+            if (actor == null || !actor.IsValid)
+                return false;
 
-                var part = avoidance.Definition.GetPart(actor.Animation);
-                if (part == null)
-                    continue;
+            var part = avoidance.Definition.GetPart(actor.Animation);
+            if (part == null)
+                return false;
 
-                var radius = Math.Max(part.Radius, actor.Radius);
-                var finalRadius = radius * avoidance.Settings.DistanceMultiplier;
-                var nodes = grid.GetNodesInRadius(actor.Position, finalRadius);
+            var radius = Math.Max(part.Radius, actor.Radius);
+            var finalRadius = radius * avoidance.Settings.DistanceMultiplier;
+            var nodes = grid.GetNodesInRadius(actor.Position, finalRadius);
 
-                if (actor.Animation != part.Animation)
-                    continue;
+            if (actor.Animation != part?.Animation)

```

Comment:
Is this necessary? part null check is already done above.

Also this will not return false when both actor and part Animations are null. Is this intended or even possibly to have null for Animation?

Reply:
Actually the whole check is redundant as they are not really used.